compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -304,6 +308,15 @@ public void visit(BLangPackage pkgNode, AnalyzerData data) {
 
         validateEnumMemberMetadata(pkgNode.constants);
 
+        // Then resolve user defined types
+        for (int i = 0; i < pkgNode.topLevelNodes.size(); i++)  {
+            // Prevent analyzing type definitions that get added while analyzing other nodes
+            if (pkgNode.topLevelNodes.get(i).getKind() == NodeKind.TYPE_DEFINITION &&
+                    i <= originalTopLevelNodeSize - 1) {
```

Comment:
Why are we doing this instead of using `originalTopLevelNodeSize` in the for loop?